packages/nimble-components/src/table/index.ts

```diff
@@ -11,7 +11,7 @@ import {
 } from '@tanstack/table-core';
 import { styles } from './styles';
 import { template } from './template';
-import type { TableRecord } from './types';
+import type { TableRecord, TableRowState, TableValidity } from './types';
```

Comment:
Do we need to export these types from `nimble-angular` [like we do for other types](https://github.com/ni/nimble/blob/86302af34ab8c23201b56001fe9efbd599e46878/angular-workspace/projects/ni/nimble-angular/src/directives/table/nimble-table.directive.ts#L6)?

Reply:
Yes, we will need to export `TableValidity` from Angular. I'll do that in a separate PR because I also need to add support for the `id-field-name` attribute, `validity` property, and `checkValidity()` function.

`TableRowState` is intended to be internal within the table even though `tableData` is required to be public in order for the template to have access to it. I don't think we need to expose that type through to Angular.